bin/crystal

```diff
@@ -141,6 +141,10 @@ CRYSTAL_DIR="$CRYSTAL_ROOT/.build"
 export CRYSTAL_PATH=lib:$CRYSTAL_ROOT/src
 export CRYSTAL_HAS_WRAPPER=true
 
+if [ -z "$CRYSTAL_LIBRARY_PATH" ]; then
+  export CRYSTAL_LIBRARY_PATH="$(crystal env CRYSTAL_LIBRARY_PATH || echo "")"
```

Comment:
Why is this needed?

If .build/crystal does not exist, the crystal binary will have the CRYSTAL_LIBRARY_PATH
If .build/crystal exists, the Makefile should have added the CRYSTAL_CONFIG_LIBRARY_PATH

Reply:
for running individual specs with `bin/crystal spec`.  The compiled spec binary does not inherit `CRYSTAL_CONFIG_LIBRARY_PATH`,  thus if there's a `run` helper call in it, it invokes the linker without the necessary arguments.

Reply:
Ok, got it.

I think is better to set `CRYSTAL_CONFIG_LIBRARY_PATH`. This way the wrapper is able to build a compiler on demand for this scenario, but we don't change the CRYSTAL_LIBRARY_PATH for every run. Since you might want to create a .build/crystal with some specific CRYSTAL_LIBRARY_PATH and compile programs with it.

```
if [ -z "$CRYSTAL_CONFIG_LIBRARY_PATH" ]; then
  export CRYSTAL_CONFIG_LIBRARY_PATH="$(crystal env CRYSTAL_LIBRARY_PATH || echo "")"
fi
```

Reply:
I can add that in #9423 if you agree. The changes are all related.

Reply:
Sure, let's condense this into #9423 :)